### PR TITLE
Implemented createSavedListing and deleteSavedListing, along with other changes!

### PR DESCRIPTION
**Pull request by arfi2212**
This pull request brings the following:

Additions/Changes:
- _createSavedListing_ has now been implemented so that a certain listing can be bookmarked by a user, for future reference.
- _deleteSavedListing_ has now been implemented so that a user can delete a previously saved listing. 
- The product class has been changed to incorporate the saved boolean variable. 
- _searchListings_ now sends in a token so that the frontend can display which products were previously saved. This also has the effect of preventing listings contributed by the user from showing up in their feed. 
- Both refreshing and filtering now work correctly on the feed. They both use productListRecall and constructNewPage to cleanly get their respective tasks done. 
- The intents on FeedRecyclerAdapter have been changed to reflect the integration of the backend. Links to the product page, the messaging page and the profile page now provide the updated information these pages require. 
- The FeedActivity navigation bar has now been moved to the right side of the screen to allow space for the future multi-page app navigation bar.

Bugs Fixed:
- The annoying null pointer bug when calculating product distances has now been painstakingly fixed (love me some latches and multithreading 😍). 
- bookmarkToggleHelper now shows the correct bookmark icon if the user had previously saved a product.

### CODE REVIEW

File: app/src/main/java/com/example/ar_reshare/SplashActivity.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ private void proceed() {
         timer.schedule(new TimerTask() {
             @Override
             public void run() {
-                Intent intent = new Intent(SplashActivity.this, ARActivity.class);
+                Intent intent = new Intent(SplashActivity.this, FallbackActivity.class);
```
Comment (artur-varosyan) [inline app/src/main/java/com/example/ar_reshare/SplashActivity.java:62]: I think this is should be AR Activity? 😄

**Review by artur-varosyan — APPROVED** [review]
Intuitive and incredibly well documented piece of code!

**Review by Lingtao-Cai — APPROVED** [review]
great job! so many endpoints done!

**Review by HellinSun — APPROVED** [review]
Bookmark function is now implemented, Great job!!!

File: app/src/main/java/com/example/ar_reshare/SplashActivity.java (hunk 1 above)
Comment (arfi2212) [reply app/src/main/java/com/example/ar_reshare/SplashActivity.java:62]: Good catch, will fix now :)

**Pull request merged**